demisto_sdk/commands/common/tests/base_validator_test.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+from demisto_sdk.commands.common.hook_validations.base_validator import \
+    BaseValidator
+from demisto_sdk.commands.common.hook_validations.structure import \
+    StructureValidator
+from demisto_sdk.tests.constants_test import (
+    INVALID_INTEGRATION_WITH_NO_TEST_PLAYBOOK, SCRIPT_WITH_PLAYBOOK,
+    VALID_INTEGRATION_TEST_PATH, VALID_TEST_PLAYBOOK_PATH)
+
+HAS_TESTS_KEY_UNPUTS = [
+    (VALID_INTEGRATION_TEST_PATH, 'integration', True),
+    (INVALID_INTEGRATION_WITH_NO_TEST_PLAYBOOK, 'integration', False)
+]
+
+
+@pytest.mark.parametrize('file_path, schema, expected', HAS_TESTS_KEY_UNPUTS)
+def test_yml_has_test_key(file_path, schema, expected):
+    structure_validator = StructureValidator(file_path, predefined_scheme=schema)
+    validator = BaseValidator(structure_validator)
+    tests = structure_validator.current_file.get('tests')
+    assert validator.yml_has_test_key(tests, schema) == expected
+
+
+FIND_TEST_MATCH_IMPUT = [
+    (
+        {'integrations': 'integration1', 'playbookID': 'playbook1'},
+        'integration1',
+        'playbook1',
+        'integration',
+        True
+    ),
+    (
+        {'integrations': 'integration1', 'playbookID': 'playbook1'},
+        'integration2',
+        'playbook1',
+        'integration',
+        False
+    ),
+    (
+        {'integrations': ['integration1', 'integration2'], 'playbookID': 'playbook1'},
+        'integration1',
+        'playbook1',
+        'integration',
+        True
+    ),
+    (
+        {'integrations': ['integration1', 'integration2'], 'playbookID': 'playbook1'},
+        'integration3',
+        'playbook1',
+        'integration',
+        False
+    ),
+    (
+        {'integrations': ['integration1', 'integration2'], 'playbookID': 'playbook1'},
+        'integration1',
+        'playbook2',
+        'integration',
+        False
+    ),
+    (
+        {'playbookID': 'playbook1'},
+        'playbook',
+        'playbook1',
+        'playbook',
+        True
+    ),
+
```

Comment:
You have `'integration'` and `'playbook'` but not script. Please add that case as well.